src/error.rs

```diff
@@ -180,6 +180,35 @@ impl<I: fmt::Display, S: fmt::Display> fmt::Display for SimpleReason<I, S> {
     }
 }
 
+/// A type representing zero, one, or many labels applied to an error
+#[derive(Clone, Copy, Debug, PartialEq)]
+enum SimpleLabel {
+    Some(&'static str),
+    None,
+    Multi,
+}
+
+impl SimpleLabel {
+    fn merge(self, other: Self) -> Self {
+        match (self, other) {
+            (SimpleLabel::Some(_), SimpleLabel::Some(_)) => SimpleLabel::Multi,
```

Comment:
I think checking for label equality here is important: without that, two `"expression"` labels would get merged into `Multi`, which is probably not desirable.